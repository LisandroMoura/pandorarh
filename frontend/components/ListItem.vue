<template>
  <div class="bg-white rounded-lg shadow-md p-4 mb-4">
    <div class="flex justify-between items-start">
      <div class="flex-1">
        <div class="text-sm text-gray-500">id: {{ id }}</div>
        <h3 class="text-lg font-semibold text-gray-800">{{ nome }}</h3>
        <p class="text-sm text-gray-600">Nascimento: {{ data_nascimento }}</p>
      </div>

      <div class="flex gap-2">
        <button @click="ver(id)"
          class="px-3 py-1 text-sm text-green-600 hover:text-green-900 border border-green-600 rounded-md transition-colors">
          Ver
        </button>
        <button @click="edit(id)"
          class="px-3 py-1 text-sm text-blue-600 hover:text-blue-800 border border-blue-500 rounded-md transition-colors">
          Editar
        </button>
        <button @click="destroy(id)"
          class="px-3 py-1 text-sm text-red-600 hover:text-red-800 border border-red-500 rounded-md transition-colors">
          Excluir
        </button>
      </div>
    </div>
  </div>
</template>

<script setup>

defineProps({
  id: Number,
  nome: String,
  data_nascimento: String
}
);

const successMessage = useState('successMessage')


const ver = (id) => {
  useRouter().push('/funcionarios/' + id);
}

const edit = (id) => {
  useRouter().push('/funcionarios/edit/' + id);
}

import { useNuxtApp } from '#app'

const destroy = async (id) => {
  const router = useRouter()
  const { authFetch } = useFetchAuth()


  if (!confirm('Tem certeza que deseja excluir este item?')) {
    return
  }

  try {

    const response = await authFetch(`http://localhost:8000/api/funcionarios/${id}`, {
      method: 'DELETE',
    });

    successMessage.value = 'Funcionário deletado com sucesso!'

    refreshNuxtData('funcionarios') // Usando a mesma chave usada no useFetch

    // Redirecionando para a lista
    useRouter().push({
      path: '/funcionarios',
    })

  } catch (error) {
    // useNuxtApp().$toast.error('Erro ao excluir funcionário');
    console.error('Erro ao excluir funcionário:', error)
  }
}


</script>

<style></style>
