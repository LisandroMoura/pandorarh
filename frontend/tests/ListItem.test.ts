import { describe, test, expect, vi, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";

import { setupMocks } from '@/tests/moks/setupMocks'

import ListItem from "@/components/ListItem.vue";

describe("ListItem", () => {
  const mockProps = {
    id: 2,
    nome: "João Silva",
    data_nascimento: "1991-01-01",
  };

  beforeEach(() => {
    setupMocks() 
    vi.clearAllMocks()
    global.confirm = vi.fn(() => true)
  })

  test("renderiza corretamente as props", async () => {
    const wrapper = await mount(ListItem, { props: mockProps });

    expect(wrapper.text()).toContain(`id: ${mockProps.id}`);
    expect(wrapper.text()).toContain(mockProps.nome);
    expect(wrapper.text()).toContain(
      `Nascimento: ${mockProps.data_nascimento}`
    );
  });

  test("renderiza os botões de ação", () => {
    const wrapper = mount(ListItem, { props: mockProps });
    const buttons = wrapper.findAll("button");
    expect(buttons).toHaveLength(3);
    // console.dir(buttons)
    expect(buttons[0].text()).toBe("Ver");
    expect(buttons[1].text()).toBe("Editar");
    expect(buttons[2].text()).toBe("Excluir");
  });  

});
