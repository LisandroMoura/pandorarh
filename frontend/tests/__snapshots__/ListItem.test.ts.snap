// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumb.vue > renderiza o breadcrumb corretamente 1`] = `
"<div>
  <!--componente custom para exibir os links de navegação: Listar e Novo-->
  <div class="w-full flex flex-col md:flex-row justify-between items-start md:items-center px-4 md:px-1 gap-4 pt-4 mb-10">
    <h1 class="text-3xl font-bold text-[#21223a] w-full md:w-auto">Funcionários da PandoraRH</h1>
    <nav class="flex gap-4 justify-start md:justify-center w-full md:w-auto"><a href="#" class="flex items-center gap-2 text-gray-900 hover:text-gray-600 px-4 py-2 transition-colors"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="size-6">
          <path stroke-linecap="round" stroke-linejoin="round" d="M8.25 6.75h12M8.25 12h12m-12 5.25h12M3.75 6.75h.007v.008H3.75V6.75Zm.375 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0ZM3.75 12h.007v.008H3.75V12Zm.375 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Zm-.375 5.25h.007v.008H3.75v-.008Zm.375 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"></path>
        </svg> Listar </a><a href="#" class="flex items-center gap-2 text-gray-900 hover:text-gray-600 px-4 py-2 transition-colors"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="size-6">
          <path stroke-linecap="round" stroke-linejoin="round" d="M12 9v6m3-3H9m12 0a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"></path>
        </svg> Novo </a></nav>
  </div>
</div>"
`;
